crates/shared/src/bad_token/token_owner_finder/blocksout.rs

```diff
@@ -0,0 +1,90 @@
+use anyhow::{bail, Result};
+use ethcontract::H160;
+use reqwest::{Client, Url};
+use serde::Deserialize;
+
+use super::TokenOwnerFinding;
+
+const BASE: &str = "https://blockscout.com/";
+
+pub struct BlockscoutTokenOwnerFinder {
+    client: Client,
+    base: Url,
+}
+
+impl BlockscoutTokenOwnerFinder {
+    pub fn try_for_network(client: Client, network_id: u64) -> Result<Self> {
+        let network = match network_id {
+            1 => "eth/",
+            100 => "xdai/",
+            _ => bail!("Unsupported Network"),
+        };
+        Ok(Self {
+            client,
+            base: Url::try_from(BASE)
+                .expect("Invalid Blockscout Base URL")
+                .join(network)
+                .expect("Invalid Blockscout URL Segement")
+                .join("mainnet/api")
+                .expect("Invalid Blockscout URL Segement"),
+        })
+    }
+}
+
+#[derive(Deserialize)]
+struct Response {
+    result: Vec<TokenOwner>,
+}
+
+#[derive(Deserialize)]
+struct TokenOwner {
+    address: H160,
+}
+
+#[async_trait::async_trait]
+impl TokenOwnerFinding for BlockscoutTokenOwnerFinder {
+    async fn find_candidate_owners(&self, token: H160) -> Result<Vec<H160>> {
+        let mut url = self.base.clone();
+        url.query_pairs_mut()
+            .append_pair("module", "token")
+            .append_pair("action", "getTokenHolders")
+            .append_pair("contractaddress", &format!("{token:#x}"));
+
+        tracing::debug!("Querying Blockscout API: {}", url);
+        let response_text = self.client.get(url).send().await?.text().await?;
+        tracing::debug!("Response from Blockscout API: {}", response_text);
+
+        let parsed = serde_json::from_str::<Response>(&response_text)?;
+        Ok(parsed
+            .result
+            .into_iter()
+            .map(|owner| owner.address)
```

Comment:
why do you need all of them? can't you do just one?
Also, if owners is empty, the empty response is handled correctly?